src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,531 @@
+     */
+    public ClientBuilder maxConnections(int maxConnections) {
+        this.maxConnections = maxConnections;
+        return this;
+    }
+
+    /**
+     * Sets a proxy url for the client connection.
+     *
+     * @param proxyURL the URL of the proxy server
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder proxyURL(URL proxyURL) {
+        this.proxyURL = proxyURL;
+        return this;
+    }
+
+    /**
+     * Sets an optional proxy username for the client connection.
+     *
+     * @param proxyUser username for the proxy server
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder proxyUser(String proxyUser) {
+        this.proxyUser = proxyUser;
+        return this;
+    }
+
+    /**
+     * Sets an optional proxy password for the proxy user specified by
+     * {@link #proxyUser(String)}.
+     *
+     * @param proxyPassword password for the proxy server user
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder proxyPassword(String proxyPassword) {
+        this.proxyPassword = proxyPassword;
+        return this;
+    }
+
+    /**
+     * Flag to disable hostname verification and certificate chain validation. This is not
+     * recommended, but for example could be useful for testing with a self-signed certificate.
+     * <P>
+     * The SSL authentication is enabled by default meaning that hostname verification
+     * and certificate chain validation is done using the JVM default settings.
+     * </P>
+     *
+     * @return this ClientBuilder object for setting additional options
+     * @throws IllegalStateException when SSLSocketFactory is not null
+     * @see #authenticatedModeSSLSocketFactory(SSLSocketFactory)
+     */
+    public ClientBuilder disableSSLAuthentication() {
+        if(authenticatedModeSSLSocketFactory == null) {
+            this.isSSLAuthenticationDisabled = true;
+        } else {
+            throw new IllegalStateException("SSLSocketFactory is not null. Please " +
+                    "create a ClientBuilder without a custom SSLSocketFactory.");
+        }
+        return this;
+    }
+    
+
+    /**
+     * Specifies the custom SSLSocketFactory to use when connecting to Cloudant over a
+     * <code>https</code> URL, when SSL authentication is enabled.
+     * <P>
+     * Note: this will have no effect if {@code disableSSLAuthentication()} is used to
+     * disable SSL authentication.
+     * </P>
+     *
+     * @param factory An SSLSocketFactory, or <code>null</code> for the
+     *                default SSLSocketFactory of the JRE.
+     * @return this ClientBuilder object for setting additional options
+     * @throws IllegalStateException when SSL authentication is disabled
+     * @see #disableSSLAuthentication()
+     */
+    public ClientBuilder authenticatedModeSSLSocketFactory(SSLSocketFactory factory) {
```

Comment:
Should we consider renaming this?
`customSSLSocketFactory`?
I think it was named this previously to indicate that the `SSLSocketFactory` only applied in the "mode" where SSL authentication was not disabled, but since these changes are making them mutually exclusive I don't think the name makes sense any more.